### DISCUSSION TITLE
How to simplify noncontinuous polygons？

### DISCUSSION THREAD

**Discussion by pzq123456** [geopandas/geopandas#2716] [Q&A]
Hello!
I am working on a program about DXF file of CAD, I wanner to simplify blocks of wall graph to a complete block.In this picture, wall blocks are noncontinuous.I do not know how to handle with them.
![output](https://user-images.githubusercontent.com/82391775/143523503-6b10c707-8626-4f9a-aff1-46b36626e473.png)
I tried the api "centroid" and would like to just connect those points to lines,but I do not know how to operate those points and union them to a whole block.
![outpu111t](https://user-images.githubusercontent.com/82391775/143523875-9bd8d15b-c918-4305-a396-6eef2da86336.png)
Thanks!

**Comment by martinfleis** [comment]
Hi, I am not sure I understand what exactly is the goal. Do you have an illustration of the final outcome you'd like to get?

**Comment by pzq123456** [comment]
![EE53841B-E303-49FD-8136-7757A0BE73B4](https://user-images.githubusercontent.com/82391775/143573632-10ecbd36-8869-45ad-8bc5-11b5cfba9d33.jpeg)

**Comment by martinfleis** [comment]
Wow, that is a hard problem if you aim to get a graph of the construction system. If you want to get a graph of rooms, that may be a tiny bit easier but not much I am afraid. 

I am now aware of any tool that can do that out of the box. You can try some combinations of buffering and snapping to close the gaps between the features but in order to get a graph you need to be able to define some clear rules on how to do that automatically and that I don't know how to do atm.
